src/main.cpp

```diff
@@ -5829,7 +5855,8 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         BlockTransactions resp;
         vRecv >> resp;
 
-        CBlock block;
+        CBlock *pblock = new CBlock();
```

Comment:
The shared_ptrs are to a const CBlock, and IIRC you cant blindly cast a shared_ptr<CBlock> to a shared_ptr<const CBlock>, but you need to access the block as non-const, and I kinda prefer this over a const_cast.